syntax = "proto3";

package roadwork.simulator;

import "system/space.proto";
import "system/space_meta.proto";

message StepResponse {
    float reward = 1; // The reward received after taking the previous action
    bool isDone = 2; // Has the episode ended?
    map<string, string> info = 3; // Diagnostic information
    system.SpaceWrapper observation = 4;
}

message ResetResponse {
    repeated double observation = 1; // The initial observation
}

message RenderResponse {}

message CloseResponse {}

message CreateResponse {
    string instanceId = 1;
}


message ActionSpaceSampleResponse {
    int32 action = 1; // Note: Serialized as bytes!
}

message ActionSpaceInfoResponse {
    system.MetaSpaceWrapper result = 1;
}

message ObservationSpaceInfoResponse {
    system.MetaSpaceWrapper result = 1;
}